hal/targets/hal/TARGET_NORDIC/TARGET_NRF5/gpio_api.c

```diff
@@ -116,7 +109,7 @@ static void gpio_apply_config(uint8_t pin)
                 if ((m_gpio_irq_enabled & (1 << pin))
                     && (m_gpio_cfg[pin].irq_rise || m_gpio_cfg[pin].irq_fall))
                 {
-                    nrf_drv_gpiote_in_event_enable(pin, false);
+                    nrf_drv_gpiote_in_event_enable(pin, true);
```

Comment:
`m_gpio_irq_enabled` is only modified when `gpio_irq_enable` or `gpio_irq_enable` is called, in that case, does it makes sense to verify `m_gpio_irq_enabled ` ? 

I don't think it change anything in the current state, that's just a question.